consensus/merge/src/test/java/org/hyperledger/besu/consensus/merge/TransitionProtocolScheduleTest.java

```diff
@@ -159,52 +155,45 @@ public void getByBlockHeader_returnsTimestampScheduleIfPresent() {
   }
 
   @Test
-  public void getByBlockNumber_returnsTimestampScheduleIfPresent() {
-    when(blockchain.getBlockHeader(BLOCK_NUMBER)).thenReturn(Optional.of(blockHeader));
-    when(timestampSchedule.getByBlockHeader(blockHeader)).thenReturn(mock(ProtocolSpec.class));
-
-    assertThat(transitionProtocolSchedule.getByBlockNumber(BLOCK_NUMBER)).isNotNull();
-
-    verify(timestampSchedule).getByBlockHeader(blockHeader);
-    verifyNoMergeScheduleInteractions();
-  }
-
-  @Test
-  public void getByBlockNumber_delegatesToPreMergeScheduleWhenBlockNotFound() {
-    when(blockchain.getBlockHeader(BLOCK_NUMBER)).thenReturn(Optional.empty());
+  public void getByBlockHeader_delegatesToPreMergeScheduleWhenBlockNotFound() {
     when(mergeContext.isPostMerge()).thenReturn(false);
 
-    transitionProtocolSchedule.getByBlockNumber(BLOCK_NUMBER);
+    transitionProtocolSchedule.getByBlockHeaderForBws(blockHeader);
 
     verifyPreMergeProtocolScheduleReturned();
   }
 
   @Test
-  public void getByBlockNumber_delegatesToPostMergeScheduleWhenBlockNotFound() {
-    when(blockchain.getBlockHeader(BLOCK_NUMBER)).thenReturn(Optional.empty());
+  public void getByBlockHeader_delegatesToPostMergeScheduleWhenBlockNotFound() {
```

Comment:
Not shown in the diff, but I think getByBlockHeader() is now only being covered by one unit test, for when the timestamp schedule is found. It needs at least the other branch of logic tested, which is similar to this unit test.

Reply:
Changed the getByBlockHeader tests to use getByBlockHeader instead of getByBlockHeaderForBws. The other tests cover the backward sync conditions.